src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -81,6 +81,33 @@ public ModificationInfos getNetworkModification(UUID networkModificationUuid) {
         return networkModificationRepository.getModificationInfo(networkModificationUuid);
     }
 
+    //get all network modifications
+    @Transactional(readOnly = true)
+    public List<ModificationInfos> getAllNetworkModifications(UUID groupUuid) {
+        List<ModificationInfos> modifications = networkModificationRepository.getAllModificationInfo(groupUuid);
+        return modifications;
+    }
+
+    @Transactional(readOnly = true)
+    // Need a transaction for collections lazy loading
+    public List<ModificationInfos> getAllNetworkModificationsWithMetaData(UUID groupUuid, boolean errorOnGroupNotFound) {
+        List<ModificationInfos> modifications = networkModificationRepository.getAllModificationInfo(groupUuid);
+        List<ModificationInfos> metaData = networkModificationRepository.getAllModificationsMetaData(groupUuid);
+        //put modification in a hashmap
+        Map<UUID, ModificationInfos> modificationInfosMap = modifications.stream().collect(Collectors.toMap(ModificationInfos::getUuid, m -> m));
+        //put metadata in a hashmap
+        Map<UUID, ModificationInfos> metaDataMap = metaData.stream().collect(Collectors.toMap(ModificationInfos::getUuid, m -> m));
+
+        modificationInfosMap.forEach((uuid,modificationInfos) -> {
+            ModificationInfos metadata = metaDataMap.get(uuid);
+            if (modificationInfos != null) {
+                modificationInfos.setStashed(metadata.getStashed());
+                modificationInfos.setDate(metadata.getDate());
+            }
```

Comment:
@SlimaneAmar  j'ai un doute sur cette implementation.
sans faire ca, l'attribut stashed est null, je me demande si il y a un autre moyen de recuperer ce status

Reply:
only in English !